src/CometWrapper.sol

```diff
@@ -92,8 +92,12 @@ contract CometWrapper is ERC4626, CometHelpers {
         if (assets == 0) revert ZeroAssets();
         if (msg.sender != owner) {
             uint256 allowed = allowance[owner][msg.sender];
-
-            if (allowed != type(uint256).max) allowance[owner][msg.sender] = allowed - shares;
+            shares = convertToShares(assets);
```

Comment:
this seems weird: having a return variable that we use to represent one value in this block, but later reassign to use as the return value.

can we delete the named return variable and then define a new variable on line 106 (and then explicitly return it at the end of the function)?

Reply:
I missed Han's comment: https://github.com/compound-finance/comet-wrapper/pull/3#discussion_r1322224413, but I agree: return variables are terrible